client/app/components/sections/SearchPage/SearchPage.js

```diff
@@ -0,0 +1,41 @@
+import { Component, PropTypes } from 'react';
+import { div, img, p } from 'r-dom';
+
+import css from './SearchPage.css';
+
+class SearchPage extends Component {
+
+  constructor(props, context) {
+    super(props, context);
+    this.listings = this.props.searchPage.get('listings') || [];
```

Comment:
This is going to be page level component. There will be methods like shouldComponentUpdate etc. - so, I decided to keep this as class from the beginning and thus make things easier to test.